src/classes/PMT_Payment_TEST.cls

```diff
@@ -381,6 +381,234 @@ private class PMT_Payment_TEST {
     /*******************************************************************************************************
      * HELPER FUNCTIONS
      *******************************************************************************************************/
+    /**
+     * @description Validate that the accounting rules are enforced properly. If there is a paid payment, it must have a payment date.
+     * If it is an unpaid payment, it must have a scheduled date. 
+     * down the Payment records.
+     */
+     @isTest
+    static void test_EnforceAccountingDataConsistency() {
```

Comment:
This is a _giant_ test method. Let's split this out into discreet test methods that test each bit of functionality separately. Each could test both insert and update.
* shouldPreventMissingPaymentDateWhenEnforceAccting()
* shouldPreventMissingScheduledDateWhenEnforceAccting()
* shouldAllowMissingPaymentDateWhenNotEnforceAccting();
* shouldAllowMissingScheduledDateWhenNotEnforceAccting();

Ideally it would be nice if you could call the `enforceAccountingDataConsistency()` method directly with a Payment record instance in the unit tests, but I'm pretty sure the `addError()` method will throw an exception if not in a trigger context.

Reply:
Done